src/App.tsx

```diff
@@ -15,49 +19,110 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  NONE,
+  ALPABET,
+  LENGTH,
+}
+
+export function getReorderedGoods(
+  goods: string[],
+  sortType: SortType,
+  isReversed: boolean,
+) {
+  // To avoid the original array mutation
+  const visibleGoods = [...goods];
+
+  // Sort and reverse goods if needed
+  visibleGoods.sort((goodA, goodB) => {
+    switch (sortType) {
+      case SortType.ALPABET:
+        return goodA.localeCompare(goodB);
+
+      case SortType.LENGTH:
+        return goodA.length - goodB.length;
+
+      default: return 0;
+    }
+  });
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
+
+  return visibleGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortType, setSortType] = useState(SortType.NONE);
+  const [isReversed, setIsReversed] = useState(false);
+  const onReorder = (sortType !== SortType.NONE || isReversed);
+  const reorderedGoods = getReorderedGoods(
+    goodsFromServer,
+    sortType,
+    isReversed,
+  );
+
+  const handleSortByAlpabet = () => setSortType(SortType.ALPABET);
+  const handleSortByLength = () => setSortType(SortType.LENGTH);
+  const handleReverse = () => (setIsReversed(value => (
+    !value
+  )));
```

Comment:
```suggestion
  const handleReverse = () => (setIsReversed(isReversed => (
    !isReversed
  )));
```

Reply:
Linter doesn't allow using a variable name "isReversed" because it was declared in the upper scope